src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1252,8 +1252,7 @@ static void uct_ib_mlx5_devx_cleanup_flush_mr(uct_ib_mlx5_md_t *md)
 {
     ucs_status_t status;
 
-    if (!(md->flags & UCT_IB_MLX5_MD_FLAG_KSM) ||
-        !uct_ib_md_is_flush_rkey_valid(md->super.flush_rkey)) {
+    if (!uct_ib_md_is_flush_rkey_valid(md->super.flush_rkey)) {
```

Comment:
Maybe we shouldn't drop `UCT_IB_MLX5_MD_FLAG_KSM` but add `UCT_IB_MLX5_MD_FLAG_KSM_ATOMIC` which would initialy on if `UCT_IB_MLX5_MD_FLAG_KSM` is on, and dropped if KSM atomic create failed?

Reply:
That's an option but IMO this flag looks redundant since there is no error path between create mr and its key. @yosefe WDYT?

Reply:
Right, it's possible KSM is supported but KSM with atomic is not supported.
@Artemy-Mellanox maybe even better test (try) it during devx md open instead of dropping during runtime?

Reply:
Yes, simple single entry KSM with atomic should be sufficient.